website/content/docs/external_plugins/_index.md

```diff
@@ -88,6 +100,46 @@ We see from `cache_ttls` that the result of `some_entry`'s `list` method will be
 
 `init` adopts the standard error conventions described in the [Errors](#errors) section.
 
+### static methods
+
+If your plugin knows some of the hierarchy and wants to describe it now rather than relying on more calls to `list` or `read` methods, it can do so by describing that hierarchy in the `methods` property of any attribute.
```

Comment:
I was thinking that the `methods` hash would be the default format, but now I'm starting to agree that defaulting to an array would be better b/c enumerating your hierarchy is an uncommon scenario and an array makes things easy for Bash plugin authors (especially if we start supporting multiple output formats). Plus, the explanation of a "null" result is a bit wordy, not very clear [e.g. what does "just like providing the method name in an array" mean?], doesn't explain why `stream`/`exec` should always return nil, and requires me to juggle two things at once (that methods can be a hash, and that methods can be an array). I also could not come up with any better documentation than what you have here, so that suggests to me that maybe the `methods` hash isn't the right approach now that we're defaulting to an array.

So, as an alternative, what do you think of something like
```
[
"list",
["read", <result_of_read>],
"exec",
...
]
```
Explaining this format would be easy and would not require an extra section. We would say something like

> You return an array of the entry's supported methods. If you're able to prefetch a method's result, then you can include the result as a tuple of `(<method>, <result>)`. Prefetching's a useful way to avoid unnecessary plugin script invocations, etc. etc. etc.

The `prefetch => use tuple-format` type of thinking would also stop us from having to explain why `stream`/`exec` (and other "dynamic" methods) would always return nil in the `methods` hash (because external plugin authors can figure out for themselves that it is impossible to prefetch `exec`/`stream`'s/dynamic method's results from their semantics.).

I don't think the validation for the tuple-extension would be too hard. Plus internally, we'd still keep the methods hash in externalPluginEntry because that is the sanest (and easiest) way to represent the data.